SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UNR-3761/SpatialTestMultipleOwnership/SpatialTestMultipleOwnership.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SpatialTestMultipleOwnership.h"
+#include "MultipleOwnershipPawn.h"
+#include "SpatialFunctionalTestFlowController.h"
+
+#include "GameFramework/PlayerController.h"
+#include "Kismet/GameplayStatics.h"
+
+/**
+ * This test automates the MultipleOwnesrhip gym which demonstrates that RPCs can be sent from multiple actors that have their owner set to
+ * a player controller. The test contains 1 Server and 2 Clients.
+ *
+ * The flow is as follows:
+ * - Setup:
+ *  - The server spawns 2 MultipleOwnershipPawns and registers them for auto-destroy.
+ *  - Every worker sets a reference to the spawned Pawns.
+ * - Test:
+ *  - Client 1 sends an RPC from both MultipleOwnershipPawns, which should be ignored.
+ *  - All workers check that the RPC was correctly ignored.
+ *  - The server sets Client 1's PlayerController to possess the first MultipleOnwershipPawn.
+ *  - Client 1 sends an RPC from both MultipleOwnershipPawns.
+ *  - All workers check that the possessed MultipleOwnershipPawn's RPC was correctly applied, whilst the unpossessed's one was ignored.
+ *  - The server sets Client 1's PlayerController to possess the second MultipleOnwershipPawn.
+ *  - Client 1 sends an RPC from both MultipleOwnershipPawns.
+ *  - All workers check that both RPCs were correctly applied.
+ *  - The server sets Client 1's PlayerController to unpossess the MultipleOwnershipPawn;
+ *  - Client 1 sends an RPC from both MultipleOwnershipPawns.
+ *  - All workers check that both RPCs were correctly applied.
+ * - Cleanup
+ *  - The MultipleOwnershipPawns are destroyed.
+ */
+
+ASpatialTestMultipleOwnership::ASpatialTestMultipleOwnership()
+	: Super()
+{
+	Author = "Andrei";
+	Description = TEXT("Test Net Reference");
+}
+
+void ASpatialTestMultipleOwnership::PrepareTest()
+{
+	Super::PrepareTest();
+
+	if (HasAuthority())
+	{
+		AddExpectedLogError(TEXT("No owning connection for actor MultipleOwnershipPawn"), 3, false);
```

Comment:
Maybe a comment above this, saying what the 3 instances of the warning we expect are.
(2 for first send, 1 for second send, etc.)